pkg/webui/components/search-panel/index.js

```diff
@@ -0,0 +1,344 @@
+
+  // Reset selected index when search results change or when switching modes.
+  useEffect(() => {
+    setSelectedIndex(0)
+  }, [searchResults, isTopEntitiesMode])
+
+  useEffect(() => {
+    const handleKeyDown = event => {
+      const listElement = listRef.current
+      let newIndex = selectedIndex
+
+      if (event.key === 'ArrowDown' || event.key === 'ArrowUp') {
+        event.preventDefault()
+        newIndex =
+          event.key === 'ArrowDown'
+            ? (selectedIndex + 1) % itemCount
+            : (selectedIndex - 1 + itemCount) % itemCount
+        setSelectedIndex(newIndex)
+
+        const item = document.getElementById(`search-item-${newIndex}`)
+        if (item) {
+          const itemThreshold = item.clientHeight
+          if (
+            item.offsetTop + item.clientHeight >
+            listElement.scrollTop + listElement.clientHeight - itemThreshold
+          ) {
+            listElement.scrollTop =
+              item.offsetTop + item.clientHeight - listElement.clientHeight + itemThreshold
+          } else if (item.offsetTop < listElement.scrollTop + itemThreshold) {
+            listElement.scrollTop = item.offsetTop - itemThreshold
+          }
+        }
+      } else if (event.key === 'Escape') {
+        onClose()
+      }
+    }
+
+    document.addEventListener('keydown', handleKeyDown)
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown)
+    }
+  }, [itemCount, items, onClose, onSelect, selectedIndex])
+
+  const handleInputKeyDown = useCallback(
+    event => {
+      if (event.key === 'Escape') {
+        onClose()
+      } else if (event.key === 'Enter') {
+        // Get DOM item and simulate click.
+        document.getElementById(`search-item-${selectedIndex}`).click()
+      }
+    },
+    [onClose, selectedIndex],
+  )
+
+  const handleQueryChange = useCallback(event => {
+    setQuery(event.target.value)
+  }, [])
+
+  let i = 0
+
+  return (
+    <div className={classNames(style.container, { 'pos-static': inline })}>
+      <div className="d-flex p-vert-cs-m p-sides-cs-xl gap-cs-s al-center">
+        <Icon className="c-icon-neutral-normal" icon={IconSearch} />
+        <input
+          className={style.input}
+          onKeyDown={handleInputKeyDown}
+          value={query}
+          onChange={handleQueryChange}
+          placeholder={formatMessage(sharedMessages.typeToSearch)}
+        />
+        <Icon className={style.xOut} icon={IconX} onClick={onClose} />
+      </div>
+      <Overlay visible={searchResultsFetching} loading spinnerMessage={sharedMessages.searching}>
+        <ScrollFader className={style.list} ref={listRef}>
+          {topEntitiesFetching && (
+            <div className={style.loading}>
+              <Spinner after={0}>
+                <Message content={m.fetchingTopEntities} />
+              </Spinner>
+            </div>
+          )}
+          {!topEntitiesFetching && noTopEntities && (
```

Comment:
```suggestion
          {!topEntitiesFetching && noTopEntities && isTopEntitiesMode && (
```

Reply:
For this scenario:
![2024-05-23_17-08_1](https://github.com/TheThingsNetwork/lorawan-stack/assets/49171937/953be338-1920-4297-bd61-9a499fafc56e)